linux_pipeline/Jenkinsfile_kernel_bisect_controller

```diff
@@ -0,0 +1,202 @@
+            name: 'JOB_TO_RUN',
+            description: 'Job to run bisect on'],
+        [$class: 'ChoiceParameterDefinition',
+            choices: 'linux-next\nazure-kernel\nnet-next',
+            name: 'KERNEL_TREE',
+            description: 'What Kernel tree to be used (Available only for upstream and developer pipelines).'],
+        [$class: 'StringParameterDefinition',
+            name: 'KERNEL_GIT_BRANCH',
+            defaultValue: 'master',
+            description: 'Branch to be built'],
+        [$class: 'StringParameterDefinition',
+            name: 'GOOD_COMMIT',
+            defaultValue: 'ad8009875f8f9d02bef7f2fb59a34b3ad323640b',
+            description: 'Good commit (this goes back)'],
+        [$class: 'StringParameterDefinition',
+            name: 'BAD_COMMIT',
+            defaultValue: '735071426eeb9631b419120f82b261d799ddde69',
+            description: 'Bad commit (can be HEAD)'],
+        [$class: 'StringParameterDefinition',
+            name: 'ENABLED_STAGES',
+            defaultValue: 'build_artifacts, publish_temp_artifacts, boot_test',
+            description: 'What stages to run. By default all stages are enabled']
+        ]
+    ]
+])
+
+node ("bisect_kernel_builder") {
+    resolveGitRepo(JOB_TO_RUN, KERNEL_TREE)
+
+    initEnv()
+
+    stage("validate_commits") {
+        validateCommit(BAD_COMMIT, "FAILURE")
+        validateCommit(GOOD_COMMIT, "SUCCESS")
+    }
+
+    bisect()
+}
+
+def resolveGitRepo(String pipeline, String gitURL) {
+
+    switch (pipeline) {
+        case "azure-pipeline":
+            pipelineName = "lis-pipeline/master";
+            kernelTree = "azure-kernel"
+            return;
+        case "upstream-pipeline":
+            pipelineName = "upstream-kernel-pipeline/master";
+            break;
+        case "developer-pipeline":
+            pipelineName = "pipeline-developer-patch-validation/master";
+            break;
+        case "bisect-commit-validator":
+            pipelineName = "pipeline-kernel-bisect-commit-validator/PR-183";
+    }
+
+    switch (gitURL) {
+        case "linux-next":
+            kernelTree = "git://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git";
+            break;
+        case "net-next":
+            kernelTree = "https://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next.git";
+            break;
+    }
+}
+
+def initEnv() {
+    withCredentials([[$class: 'StringBinding', credentialsId: ' MSFT_KERNEL_GIT_URL', variable: 'AZURE_KERNEL']]) {
+        stage ("initialize_env") {
+            def scriptFetch = sh (
+                    script: """#!/bin/bash
+                        set -xe
+
+                        if [[ ${kernelTree} == "azure-kernel" ]]; then
+                            kernel_tree=${AZURE_KERNEL}
+                        else
+                            kernel_tree=${kernelTree}
+                        fi
+
+                        mkdir -p /mnt/tmp/bisect
```

Comment:
add this value to a global variable and use it consistently: BUILD_DIR = "mnt/tmp/bisect"